### PR TITLE
Upgrade external/zip library

### PR DESCRIPTION
**Pull request by kuba--**
## Summary
Upgrade `external/zip` library to the version `f72dce22bd25d10f8575ec54cf51bf5a5fdcb5ba` (release `v0.1.30`)

Fixes: https://github.com/facebook/hermes/issues/356

### CODE REVIEW

File: external/zip/README.facebook [hunk 1]
```diff
@@ -1,11 +1,6 @@
 # Zip Library
 https://github.com/kuba--/zip
-Version: 635ef1d1452d187e027ecdcd164fa0fafa0fd2dd
-
-Delete the line in `zip.c`:
-```
-int symlink(const char *target, const char *linkpath); // needed on Linux
-```
```
Comment (kuba--) [inline external/zip/README.facebook:8]: It's not needed, anymore.

**Comment by tmikov** [comment]
I think we can import the updated version even without the CLA, since we have already imported the previous versions.

**Comment by kuba--** [comment]
> I think we can import the updated version even without the CLA, since we have already imported the previous versions. But then we will do it manually, and not through this PR.
> 
> Anyway, thanks for this great library!

BTW,  I've signed CLA.

**Comment by kuba--** [comment]
Hmm, looks like the `test-e2e` randomly fails and it's totally not related to the PR:

```
* What went wrong:
A problem occurred configuring project ':react-native-gradle-plugin'.
> Could not resolve all artifacts for configuration ':react-native-gradle-plugin:classpath'.
   > Could not resolve org.gradle.kotlin:plugins:1.3.6.
     Required by:
         project :react-native-gradle-plugin > org.gradle.kotlin.kotlin-dsl:org.gradle.kotlin.kotlin-dsl.gradle.plugin:1.3.6
      > Could not resolve org.gradle.kotlin:plugins:1.3.6.
         > Could not get resource 'https://plugins.gradle.org/m2/org/gradle/kotlin/plugins/1.3.6/plugins-1.3.6.module'.
            > Could not GET 'https://jcenter.bintray.com/org/gradle/kotlin/plugins/1.3.6/plugins-1.3.6.module'. Received status code 502 from server: Bad Gateway
   > Could not resolve org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.21.
     Required by:
         project :react-native-gradle-plugin > org.jetbrains.kotlin.jvm:org.jetbrains.kotlin.jvm.gradle.plugin:1.4.21
      > Skipped due to earlier error
```

Unfortunately, I do not have permission to re-run the job on CI.

**Comment by kuba--** [comment]
@tmikov - ok, looks like this PR is mergeable.

**Comment by kuba--** [comment]
> @tmikov has imported this pull request. If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D27058907).


Does it mean, I can close this PR?
Btw, I don't know what's the problem with _internal linter_.

**Pull request closed without merging**